src/utils/conversion.ts

```diff
@@ -2691,6 +2693,28 @@ export function toIdentityWithClaimsArray(
   }));
 }
 
+/**
+ * @hidden
+ */
+export function toIdentityWithClaimsArrayV2(
+  data: MiddlewareV2Claim[],
+  context: Context
+): IdentityWithClaims[] {
+  const groupedData = groupBy(data, 'targetId');
+
+  return Object.keys(groupedData).map(key => ({
```

Comment:
You can just use lodash `map` here and get the keys from the second argument of the callback